build-coordinator/src/main/java/org/jboss/pnc/core/builder/BuildTask.java

```diff
@@ -87,6 +80,7 @@
 
     BuildTask(BuildCoordinator buildCoordinator, 
             BuildConfiguration buildConfiguration, 
+            BuildConfigurationAudited buildConfigurationAudited,
```

Comment:
Yes, because the BuildConfigurationAudited does not contain dependency information.  So we need to at least temporarily hold on to the BuildConfiguration until we build the dependency tree of the BuildTasks.